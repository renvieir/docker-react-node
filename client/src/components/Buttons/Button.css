.button--primary {
  border-radius: 4px;
  border: 1px solid royalblue;
  background-color: white;
  width: 72px;
  height: 24px;
  text-transform: uppercase;
  color: royalblue;
  box-shadow: 0 1px #555577;
}

.button--primary:active {
  background-color: royalblue;
  color: white;
  box-shadow: none;
}