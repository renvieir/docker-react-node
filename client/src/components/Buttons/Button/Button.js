import React from 'react';
import './Button.css';

export default function Button(props) {
  return (
    <button
      className="button--primary "
      onClick={props.handleClick}
      disabled={props.disabled}
    >
      {props.children}
    </button>
  );
}
