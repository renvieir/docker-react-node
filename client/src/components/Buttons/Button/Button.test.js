import React from 'react';
import { shallow } from 'enzyme';

import toJson from 'enzyme-to-json';
import Button from './Button';

describe('Button', () => {
  let wrapper;
  const mocks = {};

  beforeEach(() => {
    wrapper = null;

    mocks.props = {
      disabled: false,
      handleClick: jest.fn(),
    }
  });

  it('should render correctly', () => {
    wrapper = shallow(<Button {...mocks.props}  />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render disabled', () => {
    mocks.props.disabled = true;
    wrapper = shallow(<Button {...mocks.props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call handleCLick', () => {
    wrapper = shallow(<Button {...mocks.props} />);
    wrapper.find('button').simulate('click');
    expect(mocks.props.handleClick).toBeCalled();
  });
});