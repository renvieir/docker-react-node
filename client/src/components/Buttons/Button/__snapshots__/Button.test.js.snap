// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<button
  className="button--primary "
  disabled={false}
  onClick={[MockFunction]}
/>
`;

exports[`Button should render disabled 1`] = `
<button
  className="button--primary "
  disabled={true}
  onClick={[MockFunction]}
/>
`;
